components/ProductInfo/middle/ShirtOptions.js

```diff
@@ -12,57 +13,83 @@ const ShirtOptions = () => {
   const [see, setSee] = useState(false);
 
   const dispatch = useDispatch();
-  const displayImg = useSelector(({ prodInfo }) => prodInfo.image);
-  const currentDisplay = displayImg;
+  const { currentImage, imageDisplayBtnArr } = useSelector(({ prodInfo }) => ({
+    currentImage: prodInfo.currentImage,
+    imageDisplayBtnArr: prodInfo.imageBtnArr,
+  }));
 
   const mouseEnter = (newImage) => {
-    dispatch(changeVisibleImage(newImage));
+    currentImage.isProd && dispatch(changeVisibleImage(newImage));
   };
 
   const mouseLeave = () => {
-    dispatch(changeVisibleImage(currentDisplay));
+    dispatch(changeVisibleImage(currentImage.url));
+  };
+  const mouseClick = (newImage) => {
+    currentImage.isProd && dispatch(changeVisibleImage(newImage));
+    currentImage.isProd && dispatch(changeCurrentImage({ url: newImage }));
+    dispatch(updImageBtnArr({ index: 0, url: newImage }));
   };
```

Comment:
On image click if the currentImage is the shirt and not sizing chart then set visibleImage to the shirtColor displayImg and the url of the currentImage to the shirtColor displayImg. Always update the first displayBtnArray (buttons on the left of the display image) to the correct color shirt